src/lib/utils/template_generator.js

```diff
@@ -128,6 +128,18 @@ class TemplateGenerator {
   getExternalProject(uri) {
     let url, folder, hgi;
     let fallback, url_fallback, folder_fallback, hgi_fallback, embarkVersion;
+    const replacements = {
```

Comment:
Only thing I'd suggest here would be to move the `replacements` out of `getExternalProject()` because everytime this function is call, we re-initialize those, which isn't needed because they are static.